From edb7ac544caf750a314090dd2926f3794e2aff9e Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 20 Oct 2023 15:38:21 +0200
Subject: [PATCH 0418/1695] core: mm: print non-secure memory ranges

Prints discovered non-secure memory ranges using debug log level.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4512d4cca6bb39d3856dcb25d73f3ac0e94adad8 core: mm: print non-secure memory ranges]
Change-Id: I1ab0668bf541a801f88a7cc70e2994bd320157b5
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/mm/core_mmu.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/core/mm/core_mmu.c b/core/mm/core_mmu.c
index 41fe9e4c1..76ae26028 100644
--- a/core/mm/core_mmu.c
+++ b/core/mm/core_mmu.c
@@ -488,6 +488,14 @@ void core_mmu_set_discovered_nsec_ddr(struct core_mmu_phys_mem *start,
 	if (!core_mmu_check_end_pa(m[num_elems - 1].addr,
 				   m[num_elems - 1].size))
 		panic();
+
+	if (CFG_TEE_CORE_LOG_LEVEL >= TRACE_DEBUG) {
+		while (num_elems--) {
+			DMSG("Non-secure memory range [%#"PRIxPA" %#"PRIxPA"]",
+			     m->addr, m->addr + m->size);
+			m++;
+		}
+	}
 }
 
 static bool get_discovered_nsec_ddr(const struct core_mmu_phys_mem **start,
-- 
2.47.2

